### ISSUE TITLE
m2m (or m2a) field on singleton collection is causing an error

### ISSUE THREAD

**Issue by martijn-dev** [directus/directus#12843]
### Describe the Bug

On opening a collection with relational fields, a call to the junction table of a relational field is done. This call contains a filter on the collection ID.
I think this ID is parsed from the URL, since on a singleton collection (where there's no ID in the URL) this filter gets filled with null.
This is causing a 500: 
```
{
  "errors": [
    {
      "message": "Cannot convert undefined or null to object",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

### To Reproduce

- Create a singleton collection.
- Add a m2m or m2a field.
- Open the collection in the frontend.

It does not matter if the field contains items or not.

### Errors Shown

```
{
  "errors": [
    {
      "message": "Cannot convert undefined or null to object",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

### What version of Directus are you using?

9.9.0

### What version of Node.js are you using?

v16.6.1

### What database are you using?

Postgres 13.3

### What browser are you using?

Chrome

### How are you deploying Directus?

running locally

**Comment by Nitwel** [comment]
This was fixed in #12831

**Comment by benhaynes** [comment]
@martijn-dev — perhaps you're seeing this one? https://github.com/directus/directus/issues/12847
